### PR TITLE
user funded ac replacement

### PR DESCRIPTION
**Pull request by evq**

### CODE REVIEW

File: datastore/grantserver/wallet.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// GetWallet by ID
+func (pg *Postgres) GetWallet(ID uuid.UUID) (*wallet.Info, error) {
+	statement := "select * from wallets where id = $1"
+	wallets := []wallet.Info{}
+	err := pg.DB.Select(&wallets, statement, ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(wallets) > 0 {
```
Comment (corymcdonald) [inline datastore/grantserver/wallet.go]: Looking at the sqlx documentation If we use `pg.DB.Get` it returns a singular item. Might be better to leverage it?

```
    // You can also get a single result, a la QueryRow
    jason = Person{}
    err = db.Get(&jason, "SELECT * FROM person WHERE first_name=$1", "Jason")
```

File: migrations/0006_create_ballots.down.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+drop table if exists ballot_issuers;
+drop table if exists ballot_creds;
```
Comment (corymcdonald) [inline migrations/0006_create_ballots.down.sql]: I think we need to rename this to be the same names as tables in the `.up.sql`

File: migrations/0006_create_ballots.up.sql [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	merchant_id text NOT NULL,
+  public_key text not null
+);
+
+create table order_creds (
+  item_id uuid primary key not null references order_items(id),
+  order_id uuid not null references orders(id),
+  issuer_id uuid not null references issuers(id);
+  blinded_creds json not null,
+  signed_creds json,
```
Comment (corymcdonald) [inline migrations/0006_create_ballots.up.sql]: Is there any risk in storing these credentials here? Do we have the DB encrypted?

File: payment/controllers.go [hunk 4]
```diff
@@ -162,3 +162,146 @@ func CreateTransaction(service *Service) handlers.AppHandler {
...
+		}
+
+		txInfo, err := service.wallet.SubmitAnonCardTransaction(r.Context(), req.WalletID, req.Transaction)
+		if err != nil {
+			return handlers.WrapError(err, "Error submitting anon card transaction", http.StatusBadRequest)
+		}
+
+		transaction, err := service.datastore.CreateTransaction(validOrderID, txInfo.ID, txInfo.Status, txInfo.DestCurrency, "anonymous-card", txInfo.DestAmount)
+		if err != nil {
+			return handlers.WrapError(err, "Error recording anon card transaction", http.StatusInternalServerError)
```
Comment (corymcdonald) [inline payment/controllers.go]: This is likely to be rare - but is there any way to rollback the anonymous card transaction if we fail to record to the db

File: migrations/0006_create_ballots.up.sql (hunk 3 above)
Comment (evq) [reply migrations/0006_create_ballots.up.sql]: no

File: payment/controllers.go (hunk 4 above)
Comment (evq) [reply payment/controllers.go]: due to the idempotency key we'll just record into the db on the next client retry

File: datastore/grantserver/wallet.go (hunk 1 above)
Comment (evq) [reply datastore/grantserver/wallet.go]: good find! can you make a follow up issue to refactor - this was just moved and I'd rather we address all such instances in one go

**Pull request merged**